### ISSUE TITLE
lastModifiedDate can be undefined

### ISSUE THREAD

**Issue by SamPotts** [TTLabs/EvaporateJS#115]
Just testing v1.0.0 and hit an issue. lastModifiedDate can be null so the `uploadKey()` method can break.

![screen shot 2016-03-08 at 3 55 08 pm](https://cloud.githubusercontent.com/assets/719092/13592456/49cb58e8-e546-11e5-94d0-f8b7523bb686.png)

**Comment by SamPotts** [comment]
Also seems to effect `createUploadFile

**Comment by tomsaffell** [comment]
@SamPotts Good catch. Mind sending a PR? Personally, even thought I love ternary statements, I think it might be better to do this as a try/catch, because in theory `lastModifiedDate` could be `in` the `file`, and yet not be a function. Obscure, sure, but if we're adding code may as well cover the bases, it's also DRYer (prevents repeating each variable name twice) . But I don't feel too strongly - I do ternaries like this fairly often, but maybe I shouldnt.. thanks.

**Comment by SamPotts** [comment]
I see what you mean, better to be safe. OK I'll add it to my other PR.

**Comment by SamPotts** [comment]
Hmm. Now I have other issues, it seems to cache the path for a failed upload file now. How do I turn off the retry after reload functionality?

**Comment by SamPotts** [comment]
Also, I noticed there's no check for localStorage support which means it'll break if users are privacy mode. It was looking so good :-(

**Comment by SamPotts** [comment]
```javascript
supported: (function() {
                if(!('localStorage' in window)) {
    				return false;
    			}

    			// Try to use it (it might be disabled, e.g. user is in private/porn mode)
                // see: https://github.com/Selz/plyr/issues/131
    			try {
                    // Add test item
    				window.localStorage.setItem('___test', 'OK');

                    // Get the test item
                    var result = window.localStorage.getItem('___test');

                    // Clean up
                    window.localStorage.removeItem('___test');

                    // Check if value matches
                    return (result === 'OK');
    			}
    			catch (e) {
    				return false;
    			}

    			return false;
            })()
```

**Comment by SamPotts** [comment]
I have updated my PR for a couple of other issues, namely:
- Issue above with lastModifiedDate not being set
- No check for localStorage support - iOS Safari will return `Error: QuotaExceededError: DOM Exception 22` if you try and set storage and the user is in private mode. I wrote a private wrapper for this which I've ported over. 
- Storing and restoring the `signParams` property. We use a token system for signing requests we were using `signParams` for this.
